include/libtorrent/piece_picker.hpp

```diff
@@ -569,19 +572,29 @@ namespace libtorrent
 			std::uint32_t piece_priority : 3;
 
 			// index in to the piece_info vector
-			std::uint32_t index;
+			prio_index_t index;
 
 #ifdef TORRENT_DEBUG_REFCOUNTS
 			// all the peers that have this piece
 			std::set<const torrent_peer*> have_peers;
 #endif
 
+#ifdef _MSC_VER
+#pragma warning(push, 1)
+#pragma warning(disable : 4268)
```

Comment:
as far as I can tell, all of these warnings are false positives and should probably be disabled at the project level